backend/app/modules/multisig_signatures/service/offchain.py

```diff
@@ -29,6 +33,30 @@ def get_last_pending_signature(
             hash=signature_db.hash,
         )
 
+    def approve_pending_signatures(self, _: Context) -> list[Signature]:
```

Comment:
the most important bit is missing - applying the changes from a signature `message`, when it's approved

Reply:
So to be sure - we'd need to add another field in the model, extract ```preparedSignature``` from the response below and apply it in the object? And are we ok to set the field as None when the message is pending? 

```{
    "created": "2023-10-27T07:34:09.184140Z",
    "modified": "2023-10-28T20:54:46.207427Z",
    "safe": "0xa40FcB633d0A6c0d27aA9367047635Ff656229B0",
    "messageHash": "0x7f6dfab0a617fcb1c8f351b321a8844d98d9ee160e7532efc39ee06c02308ec6",
    "message": "Welcome to Octant.\nPlease click to sign in and accept the Octant Terms of Service.\n\nSigning this message will not trigger a transaction.\n\nYour address\n0xa40FcB633d0A6c0d27aA9367047635Ff656229B0",
    "proposedBy": "0x5754aC842D6eaF6a4E29101D46ac25D7C567311E",
    "safeAppId": 111,
    "confirmations": [
        {
            "created": "2023-10-27T07:34:09.223358Z",
            "modified": "2023-10-27T07:34:09.223358Z",
            "owner": "0x5754aC842D6eaF6a4E29101D46ac25D7C567311E",
            "signature": "0xa37a1d5689b5daf1003a06479952701bc3574d66fa89c4433c634a864910ddf337b63354a66b11bedb5b6e4f7c0bf1fe2d2797d05dd288fb56e8e5d636a5064c1c",
            "signatureType": "EOA"
        },
        {
            "created": "2023-10-28T08:37:40.741190Z",
            "modified": "2023-10-28T08:37:40.741190Z",
            "owner": "0xa37E7b6524d312B7FABefd00F9A8e4524581Dc85",
            "signature": "0xa966dd0a074f4891a286b115adc191469bc19fe07105468ca582bd82c952165529a452e93ddbb062859bd2fd4c6efd68808a5e3444053ddd5e46244bb300c6fd1f",
            "signatureType": "ETH_SIGN"
        },
        {
            "created": "2023-10-28T20:54:46.207427Z",
            "modified": "2023-10-28T20:54:46.207427Z",
            "owner": "0x4f80Ce44aFAb1e5E940574F135802E12ad2A5eF0",
            "signature": "0x1b2ca05dbfda9d996aacf47b78f5ee6f477171c3895fe0bd496f68b33f68059463539264dffb513c4bf7857aaa646c170f3a47189a61ae9734d3724503c560f220",
            "signatureType": "ETH_SIGN"
        }
    ],
    "preparedSignature": "0x1b2ca05dbfda9d996aacf47b78f5ee6f477171c3895fe0bd496f68b33f68059463539264dffb513c4bf7857aaa646c170f3a47189a61ae9734d3724503c560f220a37a1d5689b5daf1003a06479952701bc3574d66fa89c4433c634a864910ddf337b63354a66b11bedb5b6e4f7c0bf1fe2d2797d05dd288fb56e8e5d636a5064c1ca966dd0a074f4891a286b115adc191469bc19fe07105468ca582bd82c952165529a452e93ddbb062859bd2fd4c6efd68808a5e3444053ddd5e46244bb300c6fd1f"
}```

Reply:
We can add `preparedSignature` in the model, but I don't it's necessary - we could also fetch it from the API when a message is approved

Reply:
The bigger problem might be that we need to call a correct service with correct params in order to apply the changes. To preserve that I believe we should have a `Protocol` and implement it for given services. Something similar as I did with `Verifier`